setup.py

```diff
@@ -288,6 +315,124 @@ def readme():
+        ('apply', 'a', 'Apply codestyle changes to sources.')
+    ]
+    description = "Code style check and apply (with -a)"
+    boolean_options = []
+
+    _result_marker = "Result:"
+    _project_directory_excluded = ['build', '.git']
+
+    _c_formatter = 'clang-format-6.0'
+    _c_formatter_install_msg = 'pip install clang'
+    _c_formatter_command_line = [_c_formatter, '-style=file']
+    _c_file_extensions = ['.h', '.c', '.hpp', '.cpp']
+
+    _py_checker = 'pycodestyle'
+    _py_formatter = 'autopep8'
+    _py_formatter_install_msg = 'pip install --upgrade autopep8\npip install --upgrade pycodestyle'
+    # E265 and W503 excluded because I didn't find a way to apply changes automatically
+    _py_checker_command_line = [_py_checker, '--ignore=E265,W503']
+    _py_formatter_command_line = [_py_formatter, '--in-place', '--aggressive', '--aggressive']
+    _py_file_extensions = ['.py']
+
+    def _get_file_list(self, path, search_extentions):
+        files = []
+        exluded_directories_full_path = [os.path.join(path, excluded_dir)
+                                         for excluded_dir in self._project_directory_excluded]
+
+        # r=root, d=directories, f = files
+        for r, d, f in os.walk(path):
+            # match exclude pattern in current directory
+            found = False
+            for excluded_dir in exluded_directories_full_path:
+                if r.find(excluded_dir) >= 0:
+                    found = True
+
+            if found:
+                continue
+
+            for file in f:
+                filename, extention = os.path.splitext(file)
+                if extention in search_extentions:
+                    files.append(os.path.join(r, file))
+
+        return files
+
+    def initialize_options(self):
+        self.apply = 0
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        root_dir = versioneer.get_root()
+        print("Project directory is: %s" % root_dir)
+
+        if self.apply:
+            self._c_formatter_command_line += ['-i']
+        else:
+            self._c_formatter_command_line += ['-output-replacements-xml']
+
+        import subprocess
+
+        bad_style_file_names = []
+
+        # C files handling
+        c_files = self._get_file_list(root_dir, self._c_file_extensions)
+        try:
+            for f in c_files:
+                command_output = subprocess.Popen(self._c_formatter_command_line + [f], stdout=subprocess.PIPE)
+                command_cout, command_cerr = command_output.communicate()
+                if not self.apply:
+                    if command_cout.find(b'<replacement ') > 0:
+                        bad_style_file_names.append(f)
+        except BaseException:
+            print("%s is not installed.\nPlease use: %s" % (self._c_formatter, self._c_formatter_install_msg))
+            exit(1)
+
+        # Python files handling
+        py_files = self._get_file_list(root_dir, self._py_file_extensions)
```

Comment:
Maybe you should use `pycodestyle` itself? For example, `pycodestyle hpat -q` command will return a list of files from `hpat` dir that do not satisfy for PEP8 style.

Reply:
Do you know a way to exclude `build` directory from file list provided by command `pycodestyle -q`? Because `pycodestyle hpat -q` gives the list in subdirectory only.

Reply:
Something like that: `pycodestyle -q --exclude build,examples,docs`

Reply:
Do you have an experience with `yapf`? @fschlimb requested to move to this tool. Do you think it is better to stay with `pycodestyle`?

Reply:
I don't have any experience with `yapf`.

Probably better to choose the tool with which one of you worked the longest and has significant experience. Moreover, in case the tool needs to be changed for some reason  I do not see the problem of switching to a new one.

I suggest to stay on `yapf` and move on to solving other problems :)

Reply:
> and move on to solving other problems

Completely agree!!!